Bottom: 2aad87442f67375a1d96704ceb15db2efd6f265d
Top:    1bab915ae611d41561b1d9d179e5a69378e5e88e
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-07-16 15:13:50 -0500

Create new role buildbot-worker-assets

Starts a web server for distributing assets to buildbot workers.


---
diff --git a/ansible-playbook/buildbot.yml b/ansible-playbook/buildbot.yml
index f5bf035e..b1e2fdbc 100644
--- a/ansible-playbook/buildbot.yml
+++ b/ansible-playbook/buildbot.yml
@@ -26,6 +26,25 @@
       new_user_email: buildbot@novatech-llc.com
       new_user_pass: "{{ lookup('password', '/dev/null') }}"
 
+- name: Setup Buildbot worker system
+  hosts: builder
+  become: true
+  become_user: root
+  become_method: sudo
+  tags:
+    - buildbot
+    - buildbot-worker
+  vars:
+    asset_host: "{{ hostvars['buildbot-worker-assets'].uri }}"
+    buildbot_master: "{{ hostvars['buildbot-master'].ip_addr }}"
+    container_networks:
+      - name: "{{ worker_backend_network }}"
+        links:
+          - buildbot-worker-assets:assets
+  tasks:
+  - import_role: 
+      name: buildbot-worker-assets
+
 - name: Gitlab public keys for buildbot containers
   hosts: daedalus
   become: true
diff --git a/roles/buildbot-worker-assets/defaults/main.yml b/roles/buildbot-worker-assets/defaults/main.yml
new file mode 100644
index 00000000..587e104e
--- /dev/null
+++ b/roles/buildbot-worker-assets/defaults/main.yml
@@ -0,0 +1,30 @@
+---
+
+# environment passed to the container
+container_env:
+
+# the name of the container being started
+container_name: "{{ docker_name_prefix }}{{ hostname }}"
+
+# networks of the container
+container_networks:
+- name: "{{ container_name }}-network"
+
+# port configuration of the container
+container_port_args: []
+
+# volumes mounted within the container
+container_volumes:
+- "/mnt/assets:/usr/local/apache2/htdocs/:ro"
+
+# service hostname
+hostname: "{{ role_name }}"
+
+# build arguments of the image
+image_args:
+
+# directory of the image source
+image_dir: "{{ docker_projects_dir }}/{{ image_name }}"
+
+# name of the image being built
+image_name: "httpd:alpine"
diff --git a/roles/buildbot-worker-assets/tasks/main.yml b/roles/buildbot-worker-assets/tasks/main.yml
new file mode 100644
index 00000000..410c1e54
--- /dev/null
+++ b/roles/buildbot-worker-assets/tasks/main.yml
@@ -0,0 +1,22 @@
+---
+
+- name: Create internal asset network
+  docker_network:
+    name: '{{ item.name }}'
+    state: present
+  loop: '{{container_networks}}'
+
+- name: Start asset server
+  docker_container:
+    env: "{{ container_env }}"
+    hostname: "{{ hostname }}"
+    image: "{{ image_name }}"
+    name: "{{ container_name }}"
+    networks: "{{ container_networks }}"
+    ports: "{{ container_port_args }}"
+    state: started
+    volumes: "{{ container_volumes }}"
+
+- add_host:
+    name: "{{ hostname }}"
+    uri: "http://{{ hostname }}"
